### PR TITLE
Improved pattern scanner

### PR DESCRIPTION
**Pull request by Vali0004**
Previous time:
- Total time was 1.05779s
- Per sig average was 0.0199584s
Current time:
- Total time was 0.567962s
- Per sig average was 0.0107163s
Time Differences
- Total time difference was 0.489828s
- Per sig average difference was 0.0092421
Conclusion:
The difference can be quite drastic, but I did simple changes. More optimizations could've been done but that could've hindered code readability and understandability
More can be done to get it under 0.3s to 0.4s

### CODE REVIEW

**Review by SkiddyToast — DISMISSED** [review]
LGTM, thanks Vali!

**Comment by Vali0004** [comment]
Just added another small commit to remove something unneeded that could slow it down. I'll make one final last optimization and that should be it

File: BigBaseV2/src/memory/pattern.cpp
Comment (Yimura) [inline BigBaseV2/src/memory/pattern.cpp:57]: newline

File: BigBaseV2/src/memory/range.cpp
Comment (Yimura) [inline BigBaseV2/src/memory/range.cpp]: Variables do not follow coding style. (snake_case)

File: BigBaseV2/src/memory/range.cpp
Comment (Yimura) [inline BigBaseV2/src/memory/range.cpp:107]: newline at end of function

File: BigBaseV2/src/memory/pattern.cpp
Comment (Yimura) [inline BigBaseV2/src/memory/pattern.cpp]: Does the compiler optimise this ida_sig.size() call or does it call .size() each iteration?

File: BigBaseV2/src/memory/pattern.cpp
Comment (Yimura) [inline BigBaseV2/src/memory/pattern.cpp]: Does the compiler optimise this mask.size() call or does it call .size() each iteration?

File: BigBaseV2/src/memory/range.cpp
Comment (Yimura) [inline BigBaseV2/src/memory/range.cpp]: Does the compiler optimise this m_size - length or does it calculate this difference each iteration?

File: BigBaseV2/src/memory/batch.cpp
Comment (Yimura) [inline BigBaseV2/src/memory/batch.cpp:26]: Looks ugly but will do for now.

File: BigBaseV2/src/logger.hpp
Comment (Yimura) [inline BigBaseV2/src/logger.hpp:98]: newline before this

**Review by Yimura — CHANGES_REQUESTED** [review]
Thanks for improving the sig scanner. 

Still a few things that require changes before we can merge them into master.

File: BigBaseV2/src/memory/pattern.cpp
Comment (Vali0004) [reply BigBaseV2/src/memory/pattern.cpp:57]: Fixed in newest commit

File: BigBaseV2/src/memory/range.cpp
Comment (Vali0004) [reply BigBaseV2/src/memory/range.cpp]: Will commit soon, waiting for review to see if there was anything else.

File: BigBaseV2/src/memory/range.cpp
Comment (Vali0004) [reply BigBaseV2/src/memory/range.cpp]: I believe it does, however to be safe, I will add it to a const var

File: BigBaseV2/src/memory/batch.cpp
Comment (Vali0004) [reply BigBaseV2/src/memory/batch.cpp:26]: True

File: BigBaseV2/src/logger.hpp
Comment (Vali0004) [reply BigBaseV2/src/logger.hpp:98]: Fixed in soon commit

File: BigBaseV2/src/memory/pattern.cpp
Comment (Vali0004) [reply BigBaseV2/src/memory/pattern.cpp]: It should, let me change

File: BigBaseV2/src/memory/pattern.cpp
Comment (Vali0004) [reply BigBaseV2/src/memory/pattern.cpp]: Should do so, commiting changes now.

Comment (Vali0004) [reply BigBaseV2/src/memory/pattern.cpp]: Committed requested changes.

File: BigBaseV2/src/memory/pattern.cpp
Comment (Vali0004) [reply BigBaseV2/src/memory/pattern.cpp]: Committed requested changes.

File: BigBaseV2/src/memory/range.cpp
Comment (Vali0004) [reply BigBaseV2/src/memory/range.cpp:107]: Done

**Comment by Vali0004** [comment]
Changes should've been resolved, please re-review everything and ensure they were done as asked.

**Review by Yimura — DISMISSED** [review]

File: BigBaseV2/src/memory/range.cpp
Comment (Vali0004) [inline BigBaseV2/src/memory/range.cpp:58]: https://uploadi.ng/uTEG0LGC

**Review by Yimura — APPROVED** [review]

**Pull request merged**
